include/frozen/bits/prg.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Frozen
+ * Copyright 2016 QuarksLab
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef FROZEN_LETITGO_PRG_H
+#define FROZEN_LETITGO_PRG_H
+
+#include <cstdint>
+
+namespace frozen {
+namespace bits {
+
+// Parameters taken from
+// https://en.wikipedia.org/wiki/Linear_congruential_generator
+// Donald Knuth, MMIX generator
```

Comment:
Should we have a ``<random>`` header in frozen that impelments constexpr version of a few standard PRNG?

Reply:
yeah, that would be cool :)

Reply:
Btw: the only reason i picked knuth mmix generator from that list is:
- it says it has a period of 2^64, which is good, it means all seed values are possible outputs iiuc
- it doesnt say that only the first 32 bits are good or anything, many of the other generators say that

In the multiply shift hasher its natural to use full 64 bit seed to multiply, so that seemed like a plus.

There may be some reason that its better to use 2 draws from 32 bit generator and put them together instead of this one, i dont really know